File: 2530 인공지능 시계.c
#define _CRT_SECURE_NO_WARNINGS
#include<stdio.h>

int main() {

    int h, m, s, takes;
    int newh, newm, news;
    int carryh = 0, carrym = 0;

    scanf("%d %d %d", &h, &m, &s);
    scanf("%d", &takes);

    if (s + takes <= 59) {
        printf("%d %d %d", h, m, s + takes);
    }
    else {
        news = s + takes;

        while (news > 59) {
            news -= 60;
            carrym++;
        }

        newm = m + carrym;
        if (newm <= 59) {
            printf("%d %d %d", h, newm, news);
        }
        else {
            while (newm > 59) {
                newm -= 60;
                carryh++;
            }

            newh = h + carryh;
            while (newh > 23) {
                newh -= 24;
            }
            printf("%d %d %d", newh, newm, news);
        }

    }

    return 0;
}

File: 2530 인공지능 시계.py
h, m, s = map(int, input().split())
takes = int(input())

carryh = 0
carrym = 0

if s + takes <= 59:
    print("%d %d %d" % (h, m, s + takes))
else:
    news = s + takes
    while news > 59:
        news -= 60
        carrym += 1

    newm = m + carrym
    if newm <= 59:
        print("%d %d %d" % (h, newm, news))
    else:
        while newm > 59:
            newm -= 60
            carryh += 1

        newh = h + carryh
        while newh > 23:
            newh -= 24
        print("%d %d %d" % (newh, newm, news))
